### ISSUE TITLE
Add additional alerting effects to minefield warnings

### ISSUE THREAD

**Issue by cmbasnett** [DarklightGames/DarkestHour#499]
Player "Rincewind" suggested we add a sound effect to accompany the out-of-bounds warnings. I'm for this, as the warnings are sometimes easy to miss if you're focusing on something else or if there's a lot going on.

Other games like the Battlefield series make your screen 'red-out' until you are out of the no-go area. We could perhaps do something similar. A not-so-subtle warning sound would be the quickest option, as it would plug into the code very easily.

**Comment by dynamittt** [comment]
I totally disagree to this, I am against spicing the mod with mainstream stuff. I like how they did it in RO2 though.... (without the visual effect): the game plays a warning voice from your comrade:" Where are you going? Return to the battlefield or you will be shot as a deserter etc. etc."

**Comment by AndrewTheel** [comment]
Perhaps the best option would be to use the blur effect, because it's part of the game to indicate you are being shot at or are under dire circumstances.  Which being in a minefield is pretty dire.

**Comment by handsm** [comment]
I think this is a good idea in general. People can miss the minefield warnings if they are busy and focusing on something else, or even if they are driving and not really focusing.

I'm not personally too fussed what kind of warning method is used. I don't think any of the methods discussed would detract from this game, as it's just a way of making sure the player knows he needs to get out of there - it's a place he shouldn't be going. Sometimes a method that's been used in a game that's more of an arcade shooter can be fine for us too.

I do think we would need to have different treatment for minefields that can activate with the player inside them, e.g. spawn protection minefields. Fair enough to red/blur/sound warn a player who is moving into a minefield at the edge of the map or in any place he should never go, because we just want him to turn around quickly and get out of there. But if he finds himself in a newly activated spawn protection minefield then he hasn't done anything wrong, but may have to walk or drive some way to get clear. We musn't interfere with his game during that period. I wrote a spawn protection subclass of the DH mine volume actor, which has major functionality advantages and I think ought to be used for all minefields that activate when a spawn becomes active. We could easily differentiate warning effects in that subclass.
